### ISSUE TITLE
Crash from [IBGCache readModelsFromPath:]

### ISSUE THREAD

**Issue by LFabien** [Instabug/Instabug-iOS#244]
We are getting more and more crash reports from our customers. Our app crashes after being open about 30 seconds. The reports show that it is from `[IBGCache readModelsFromPath:]`. Deleting and reinstalling the app doesn't fix the issue so it has a big impact on our customers.

### Stack trace

**Raw**
[ibcache_crash.txt](https://github.com/Instabug/Instabug-iOS/files/1968243/ibcache_crash.txt)

<img width="960" alt="crashlytics" src="https://user-images.githubusercontent.com/10099/39537756-613ba884-4def-11e8-97a2-c23db0f4a796.png">

### More information

A month ago, we were using Instabug v7.3 and we received some reports that our app would crash after 30 seconds of use. I installed a build to one of our impacted user and I was able to see that `InstabugDataModel.sqlite` would grow very quickly until taking all memory and then crash the app.

<img width="300" alt="screen shot 2018-05-02 at 11 05 02 am" src="https://user-images.githubusercontent.com/10099/39540830-b6bac656-4df8-11e8-8af4-b0976c8bd0ee.png">

I updated our app with Instabug v7.12.0 and it fixed the issue for this user.  However, I would still get a lot of crash reports from InstabugCore:

<img width="509" alt="screen shot 2018-05-02 at 10 56 09 am" src="https://user-images.githubusercontent.com/10099/39540552-d112f25e-4df7-11e8-995a-4003bda101d3.png">

Then, I updated our app to use Instabug to v7.12.5 yesterday and we're still getting crashes on `[IBGCache readModelsFromPath:]`.

Please let know if you need anything else.

### Instabug integration code

```
    [Instabug startWithToken:@"X" invocationEvent:IBGInvocationEventScreenshot];
    [Instabug setCrashReportingEnabled:NO];
    [Instabug setIntroMessageEnabled:NO];
    [Instabug setPromptOptionsEnabledWithBug:YES feedback:YES chat:NO];
```

### SDK Version

v7.12.5 (but previous versions have the same issue)

### iOS Version

iOS 11.3 and earlier. We asked our users to install the latest versions of iOS but that didn't fix the issue.

### Device Type

All iPads.

**Comment by Kmohamed** [comment]
@LFabien We are very sorry for inconvenience. I will go through each crash.
we have three issues here
1- [IBGCache readModelsFromPath:] I will investigate inside it and get back to you very soon
2- InstabugDataModel.sqlite which is fixed
3- The last crash it should be fixed on the latest version.

**Comment by LFabien** [comment]
@Kmohamed Do you have any news about this issue?

**Comment by Kmohamed** [comment]
We are going to release a new version today 
I will inform once we release it Most probably within couple of hours 
Sorry for being late.

**Comment by Kmohamed** [comment]
@LFabien we are very sorry for being late but finally we released a fix for this crash in SDK version 7.12.6

**Comment by Kmohamed** [comment]
Let me check it and grt back to you soon.

**Comment by Kmohamed** [comment]
@LFabien could you send to me the methods that you use from Instabug ?

**Comment by LFabien** [comment]
@Kmohamed 

*Setup*
```
    [Instabug startWithToken:@"X" invocationEvent:IBGInvocationEventScreenshot];
    [Instabug setCrashReportingEnabled:NO];
    [Instabug setIntroMessageEnabled:NO];
    [Instabug setPromptOptionsEnabledWithBug:YES feedback:YES chat:NO];
    [Instabug setSessionProfilerEnabled:NO];

    [Instabug setPreSendingHandler:^{
        [DDLog flushLog];
        [Instabug clearFileAttachments];
        // Attach last 2 days
        [Instabug addFileAttachmentWithURL:[NSURL fileURLWithPath:weakSelf.fileLogger.currentLogFileInfo.filePath]];
        NSArray<NSString *> *logs = weakSelf.fileLogger.logFileManager.sortedLogFilePaths;
        if (logs.count > 1) {
            [Instabug addFileAttachmentWithURL:[NSURL fileURLWithPath:logs[1]]];
        }
    }];
```

*Logs & Errors*
```
    [Instabug reportError:error];
    [Instabug logVerbose:logMsg]; and similar methods
    [Instabug IBGLog:logMsg];
```
*Invocation*
```
    [Instabug invokeWithInvocationMode:IBGInvocationModeNewBug];
```

**Comment by LFabien** [comment]
If that can help, I also see a few crashes from:

`-[IBGCache writeModels:toPath:]`
[logs.txt](https://github.com/Instabug/Instabug-iOS/files/2018824/writeModels.txt)

`IBGCoreDataPersistenceManager.m line 113`
[logstxt](https://github.com/Instabug/Instabug-iOS/files/2018828/coredata.txt)

**Comment by Kmohamed** [comment]
Thanks a lot definitely will be help but could you send to me the exception name ?

**Comment by LFabien** [comment]
`[IBGCache readModelsFromPath:].`
<img width="624" alt="screen shot 2018-05-18 at 4 34 23 pm" src="https://user-images.githubusercontent.com/10099/40262047-5f9e55ea-5ab9-11e8-8b44-87dd288518ae.png">

`-[IBGCache writeModels:toPath:]`
<img width="558" alt="screen shot 2018-05-18 at 4 33 16 pm" src="https://user-images.githubusercontent.com/10099/40262023-3a6d34c6-5ab9-11e8-9b1d-9ca2e071af3d.png">

`IBGCoreDataPersistenceManager`
<img width="782" alt="screen shot 2018-05-18 at 4 34 08 pm" src="https://user-images.githubusercontent.com/10099/40262086-a0864cfc-5ab9-11e8-98fc-7149a155ff47.png">

**Comment by Kmohamed** [comment]
@LFabien Thanks a lot  🙏

**Comment by Kmohamed** [comment]
I wanna inform you that I am still on it and will keep you updated once I fix it.
Thanks @LFabien

**Comment by Kmohamed** [comment]
Btw are you watching WWDC ? 😅

**Comment by LFabien** [comment]
Haha yes I am :)

**Comment by LFabien** [comment]
I have received a few more reports of users saying that our app crashes after ~20 seconds. Are you aware that this issue is still happening and is it going to be fixed in the next release? 

FYI We're using`v7.13` and the users don't have this issue with a build without Instabug.  

Suggestion: when releasing a new version, it would be great if you could add the closed Github issues in your release notes.  It would be easier to follow what's happening than `Various bug fixes`. 

Thanks again @Kmohamed

**Comment by Kmohamed** [comment]
@LFabien I don’t understand you are you talking about the same stack trace in this issue or another crash ?
totally agree about issues number will do. 👍

**Comment by LFabien** [comment]
@Kmohamed I'm not sure if it's related because I don't have any stack trace for this issue. 
It seems to be similar to the crash with `InstabugDataModel.sqlite` which was supposed to be fixed. I can open a new issue if you want.

**Comment by Kmohamed** [comment]
@LFabien Yes please and I will investigate again in this issue.

**Comment by Kmohamed** [comment]
@LFabien This crash is fixed in [7.14.3](https://github.com/Instabug/Instabug-iOS/releases/tag/7.14.3)
Thanks a lot 🙏🙏
